optuna_dashboard/_app.py

```diff
@@ -255,6 +262,29 @@ def save_study_note(study_id: int) -> dict[str, Any]:
         response.status = 204  # No content
         return {}
 
+    @app.post("/api/studies/<study_id:int>/preference")
+    @json_api_view
+    def post_preference(study_id: int) -> dict[str, Any]:
+        try:
+            best_trials = [int(d) for d in request.json.get("best_trials", [])]
+            worst_trials = [int(d) for d in request.json.get("worst_trials", [])]
+        except ValueError:
+            response.status = 400
+            return {"reason": "best_trials and worst_trials must be an array of integers."}
+        if len(best_trials) == 0 or len(worst_trials) == 0:
+            response.status = 400  # Bad request
+            return {"reason": "You need to set best_trials and worst_trials"}
+
+        try:
+            preferences = [(best, worst) for best in best_trials for worst in worst_trials]
+            report_preferences(study_id, storage, preferences)
+        except Exception as e:
+            response.status = 500
+            return {"reason": f"Internal server error: {e}"}
```

Comment:
nit: You don't need to catch exceptions since `@json_api_view` decorator can handle unexpected exceptions and returns the error message.

https://github.com/optuna/optuna-dashboard/blob/852dd3d16b2679100860961b839803b7ee28a901/optuna_dashboard/_bottle_util.py#L26-L43

```suggestion
        preferences = [(best, worst) for best in best_trials for worst in worst_trials]
        report_preferences(study_id, storage, preferences)
 ```

Reply:
I noticed that there are some api views that handles unexpected exceptions. We can fix them in the following pull reuqests.

https://github.com/optuna/optuna-dashboard/blob/852dd3d16b2679100860961b839803b7ee28a901/optuna_dashboard/_app.py#L317-L339